### PR TITLE
Add isSystemReservedApp flag to wso2carbon-local-sp

### PR DESCRIPTION
**Pull request by bhagyasakalanka**
### Proposed changes in this pull request

Need to add a flag to separately identify wso2carbon-local-sp. Adding isSystemReservedApp config to the wso2carbon-local-sp create and update flows

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java [hunk 1]
```diff
@@ -1692,6 +1697,21 @@ public void updateLocalSp(HttpServletRequest request) {
         serviceProvider.setApplicationName(request.getParameter("spName"));
         serviceProvider.setDescription(request.getParameter("sp-description"));
 
+        // Check if isSystemDefaultApp property is already set.
+        boolean isSystemDefaultAppPropExist = false;
+        for (ServiceProviderProperty property : serviceProvider.getSpProperties()) {
```
Comment (bhagyasakalanka) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java:1702]: Check if getSpProperties can be null

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationConstants.java [hunk 2]
```diff
@@ -98,6 +98,10 @@ private ApplicationConstants() {
 
     // Application Management Service Configurations.
     public static final String ENABLE_APPLICATION_ROLE_VALIDATION_PROPERTY = "ApplicationMgt.EnableRoleValidation";
+    // System Default App Configs
+
+    public static final String IS_SYSTEM_DEFAULT_APP = "isSystemDefaultApp";
```
Comment (bhagyasakalanka) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationConstants.java:103]: Add a comment explaining properties

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/util/ApplicationMgtUIConstants.java [hunk 3]
```diff
@@ -46,6 +46,10 @@ private ApplicationMgtUIConstants() {
     public static final String DEFAULT_ITEM_VALUE = "service_providers_list";
     public static final String PAGINATION_VALUE = "region=%s&item=%s";
     public static final String PAGINATION_VALUE_WITH_FILTER = "region=%s&item=%s&filterString=%s";
+    public static final String IS_SYSTEM_DEFAULT_APP = "isSystemDefaultApp";
+    public static final String IS_SYSTEM_DEFAULT_APP_DISPLAY_NAME = "Is System Default Application";
+    public static final String LOCAL_SP = "wso2carbon-local-sp";
```
Comment (bhagyasakalanka) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/util/ApplicationMgtUIConstants.java:51]: Add the constant to common and read from there

Comment (bhagyasakalanka) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/util/ApplicationMgtUIConstants.java:51]: use `isSystemReservedApp` as the flag name

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java (hunk 1 above)
Comment (bhagyasakalanka) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/ApplicationBean.java:1703]: SP properties are initialized when creating the ServiceProvider object. So no need to check for null. ref: https://github.com/wso2/carbon-identity-framework/blob/eece0de2e5793bdcc43658b81434df71bb1788f5/components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/ServiceProvider.java#L101

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationConstants.java (hunk 2 above)
Comment (bhagyasakalanka) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationConstants.java]: fixed 0e486db9380ed3167b7e5cebdde304bc8e07c28d

File: components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/util/ApplicationMgtUIConstants.java (hunk 3 above)
Comment (bhagyasakalanka) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt.ui/src/main/java/org/wso2/carbon/identity/application/mgt/ui/util/ApplicationMgtUIConstants.java]: Fixed 0e486db9380ed3167b7e5cebdde304bc8e07c28d

**Review by sachiniWettasinghe — APPROVED** [review]

**Pull request merged**